doc/sphinx/practical-tools/utilities.rst

```diff
@@ -145,7 +145,12 @@ An optional file ``CoqMakefile.local`` can be provided by the user in order to
 extend ``CoqMakefile``. In particular one can declare custom actions to be
 performed before or after the build process. Similarly one can customize the
 install target or even provide new targets. Extension points are documented in
-paragraph :ref:`coqmakefilelocal`.
+paragraph :ref:`coqmakefilelocal`. If the user needs to access variables
+defined in the ``Makefile`` in a way not covered by `secondary expansion
+<https://www.gnu.org/software/make/manual/html_node/Secondary-Expansion.html>`_,
+for example to add new dependencies to compiled outputs, the optional file
+``CoqMakefile.local-late`` can be used.  A non-exhaustive list is documented in
+paragraph :ref:`coqmakefilelocallate`.
```

Comment:
```suggestion
install target or even provide new targets. See
:ref:`coqmakefilelocal`. To access variables
defined in the ``Makefile`` in a way not covered by `secondary expansion
<https://www.gnu.org/software/make/manual/html_node/Secondary-Expansion.html>`_,
for example to add new dependencies to compiled outputs, use the optional file
``CoqMakefile.local-late``.  See :ref:`coqmakefilelocallate` for a non-exhaustive list.
```

Reply:
This change results in a switch from passive voice to active voice in the middle of the paragraph.  I'm inclined to leave the addition as-is here.

Reply:
(Also, the change you suggest makes it less clear, I think, what explanation the reader should expect by following the paragraph references.)

Reply:
I will leave it up to you.

Nonetheless, while the passive voice is commonly used in academic writing, we're aiming for a more readable style.  The preceding sentences would read better as:

> Use the optional file ``CoqMakefile.local`` to
extend ``CoqMakefile``. In particular, you can declare custom actions to
perform before or after the build process.

This is wordy: "If the user needs to access variables"
Hyperlinks should just be "See ..."; "documented in paragraph :ref:`coqmakefilelocallate`." is wordy: everything in the file is documentation and it's not necessary to say that it's a paragraph.I will leave it up to you.

Nonetheless, while the passive voice is commonly used in academic writing, we're aiming for a more readable style.  The preceding sentences would read better as:

> Use the optional file ``CoqMakefile.local`` to
extend ``CoqMakefile``. In particular, you can declare custom actions to
perform before or after the build process.

This is wordy: "If the user needs to access variables"
Hyperlinks should just be "See ..."; "documented in paragraph :ref:`coqmakefilelocallate`." is wordy: everything in the file is documentation and it's not necessary to say that it's a paragraph.

Reply:
Updated, please take a look